dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPTests.java

```diff
@@ -768,6 +768,37 @@ public void testPI59436() throws Exception {
         this.verifyStringInResponse(PI59436_APP_NAME, "PI59436.jsp", "Test passed.");
     }
 
+    /**
+     * Verify TLD file check per issue 18411.
+     * Run with applicationManager autoExpand="false" (default)
+     *
+     * @throws Exception
+     */
+    @Test
+    @Mode(TestMode.FULL)
+    public void testTLD() throws Exception {
+        // Use TestEDR app but just call index.jsp twice.
+        // 2nd call should not have SRVE0253I message if issue 18411 is fixed
+        // and no other files included in the JSP are updated.
+        String orgEdrFile = "headerEDR1.jsp";
+        String relEdrPath = "../../shared/config/ExtendedDocumentRoot/";
+        server.copyFileToLibertyServerRoot(relEdrPath, orgEdrFile);
+        String url = JSPUtils.createHttpUrlString(server, TestEDR_APP_NAME, "index.jsp");
+        LOG.info("url: " + url);
+        server.setMarkToEndOfLog();
+        WebConversation wc1 = new WebConversation();
+        WebRequest request1 = new GetMethodWebRequest(url);
+        wc1.getResponse(request1);
+
+        Thread.sleep(5000L);
```

Comment:
Maybe add a comment to explicitly state a minimum of 5 seconds is needed for this test.

Reply:
added comment